src/sdl_ui.cpp

```diff
@@ -156,9 +163,9 @@ SdlUi::SdlUi(long width, long height, const std::string& title, bool fs_flag) :
 	ShowCursor(false);
 #endif
 
-#if defined(HAVE_SDL_MIXER)
+#ifdef HAVE_SDL_MIXER
 	audio_.reset(new SdlAudio());
-#elif defined(HAVE_OPENAL)
+#elif HAVE_OPENAL
```

Comment:
These checks are not the same. You check if `HAVE_SDL_MIXER` is defined at all (may be set `0` => disabled or even some random string `"foobar"`) and then check if `HAVE_OPENAL` is defined _and_ set to true (i.e. `1`).
I do not consider the way it was done before really better, but it should be consistent.